libecs/source/Core/SceneManager.cxx

```diff
@@ -1,20 +1,14 @@
 #include "pivot/ecs/Core/SceneManager.hxx"
 
-void SceneManager::Init()
-{
-    _levels.clear();
-    _currentActiveLevel = LevelId(-1);
-}
-
 LevelId SceneManager::registerLevel(std::string name)
 {
-    _levels[LevelId(_levels.size())] = Scene(name);
+    _levels.insert({_levels.size(), std::make_unique<Scene>(name)});
```

Comment:
Why store the level into a map if you will only access it with it index? Sound like a `std::vector` with extra steps to me.